### ISSUE TITLE
Custom processing for HTTP error response

### ISSUE THREAD

**Issue by iamjoyce** [mauron85/cordova-plugin-background-geolocation#450]
## Your Environment
* Plugin version: 2.3.5
* Platform: iOS and Android
* OS version: Windows
* Device manufacturer and model: -
* Cordova version (`cordova -v`): 8.0.0
* Cordova platform version (`cordova platform ls`): android 7.0.0, ios 4.5.4
* Plugin configuration options: 
* Link to your project:

## Actual Behavior

If server response for [HTTP requests](https://github.com/mauron85/cordova-plugin-background-geolocation#custom-post-template) is 500, the library will automatically assume that server request failed.

## Context

Is there any way to change how the HTTP error handling works?

Basically I have to consume an API that I have absolutely no control in. Error response is returned in HTTP status `200` and the `body` content needs to be checked in order to determine whether the request is successful or unsuccessful.

**Comment by iamjoyce** [comment]
Most likely no
